core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -123,8 +105,21 @@ public URI getLocation() {
   }
 
   @Override
-  public JournalReader getReader() {
-    return new UfsJournalReader(this);
+  public JournalReader getReader(JournalReaderOptions options) {
+    if (options.getLocation() != null) {
+      return new UfsJournalFileReader(options);
+    } else {
+      return new UfsJournalReader(this, options);
+    }
+  }
```

Comment:
`UfsJournalReader` and `UfsJournalFileReader` are two separate concepts, they shouldn't be obtained through the same `getReader` call